index.d.ts

```diff
@@ -131,31 +138,165 @@ declare module 'warframe-items' {
+        category: 'Primary';
+    }
+    interface ExaltedWeapon extends Omit<Gun, 'category'>, Omit<Melee, 'category'>, Omit<Weapon, 'category'> {
+        category: 'Misc';
+        type: 'Exalted Weapon';
+    }
+    interface Weapon extends Equippable, Buildable, Attackable, WikiaItem, BaseItem {
+        category: Gun['category'] | Melee['category'];
+        fusionLimit?: number;
+        secondsPerShot?: number;
+        damagePerSecond?: number;
+        trigger?: Trigger;
+        damageTypes?: DamageTypes;
+        areaAttack?: AreaAttack;
+        secondary?: Secondary;
+        secondaryArea?: SecondaryArea;
+        statusChance?: number;
+        multishot?: number;
+        attacks?: Attack[];
         maxLevelCap?: number;
-        modSetValues?: number[];
-        excludeFromCodex?: boolean;
+    }
+    interface Arcane extends Buildable, Partial<Omit<Mod, 'category' | 'type'>> {
+        category: 'Arcanes';
+        type: 'Arcane';
+        drops?: Drop[];
         levelStats?: LevelStat[];
-        introduced?: Update;
-        attacks?: Attack[];
-        i18n?: I18nBundle<UniqueName>;
-        isPrime?: boolean;
     }
-
-    interface Component {
-        uniqueName: string;
-        name: string;
-        description?: string;
+    interface StanceMod extends Omit<Mod, 'levelStats'> {
+        type: 'Stance Mod';
+    }
+    interface RailjackMod extends Omit<Mod, 'levelStats'> {
+        upgradeEntries?: UpgradeEntry[];
+        category: 'Mods';
+        type: 'Railjack Mod';
+    }
+    interface RivenMod extends Omit<Mod, 'levelStats' | 'type' | 'category' | 'polarity'> {
+        category: 'Mods' | 'Misc';
+        type: 'Riven Mod' | `${WeaponType} Riven Mod`;
+        availableChallenges?: AvailableChallenge[];
+        upgradeEntries?: UpgradeEntry[];
+        polarity?: Polarity;
+    }
+    interface SingleLevelMod extends Omit<Mod, 'levelStats'> {}
+    interface Mod extends MinimalItem, WikiaItem, Droppable {
+        baseDrain: number;
+        category: 'Mods';
+        compatName?: string;
+        fusionLimit: number;
+        isAugment?: boolean;
+        isPrime: false;
+        levelStats?: LevelStat[];
+        polarity: Polarity;
+        transmutable?: boolean;
+        /** @deprecated probably replaced by Mod#isExilus */
+        isUtility?: boolean;
+        modSet?: UniqueName;
+        isExilus?: boolean;
+        modSetValues?: number[];
+        type: 'Railjack Mod' |
+            'Necramech Mod' |
+            'Warframe Mod' |
+            'Secondary Mod' |
+            'Melee Mod' |
+            'Companion Mod' |
+            'Primary Mod' |
+            'K-Drive Mod' |
+            'Riven Mod' |
+            'Archwing Mod' |
+            'Arch-Melee Mod' |
+            'Arch-Gun Mod' |
+            'Shotgun Mod' |
+            'Creature Mod' |
+            'Stance Mod' |
+            'Parazon Mod' |
+            'Transmutation Mod' |
+            'Peculiar Mod';
+    }
```

Comment:
I think these long union types are better outside the interface, for aesthetics and easy to reuse.
```suggestion
    type ModType = 
        'Railjack' |
        'Necramech' |
        'Warframe' |
        'Secondary' |
        'Melee' |
        'Companion' |
        'Primary' |
        'K-Drive' |
        'Riven' |
        'Archwing' |
        'Arch-Melee' |
        'Arch-Gun' |
        'Shotgun' |
        'Creature' |
        'Stance' |
        'Parazon' |
        'Transmutation' |
        'Peculiar';
    interface Mod extends MinimalItem, WikiaItem, Droppable {
        baseDrain: number;
        category: 'Mods';
        compatName?: string;
        fusionLimit: number;
        isAugment?: boolean;
        isPrime: false;
        levelStats?: LevelStat[];
        polarity: Polarity;
        transmutable?: boolean;
        /** @deprecated probably replaced by Mod#isExilus */
        isUtility?: boolean;
        modSet?: UniqueName;
        isExilus?: boolean;
        modSetValues?: number[];
        type: `${ModType} Mod`;
    }
```